docs/ref/models/fields.txt

```diff
@@ -808,6 +808,12 @@ Any combination of these options will result in an error.
     instead of a ``DateField`` and deciding how to handle the conversion from
     datetime to date at display time.
 
+.. warning::
+
+    A ``datetime.datetime`` instance used with :class:`DateField` will be
+    localized to the :ref:`default timezone <default-current-time-zone>` and
+    converted to a ``datetime.date`` instance, removing its time component.
```

Comment:
Sorry was rechecking the advice and

> Ideally, a symmetrical comment should be added in the documentation of DateField to explain what happens when you misuse it with a datetime.datetime values.

I think we can improve the symmetry between the two, something like

```suggestion
.. warning:: Always use :class:`DateField` with a ``datetime.date`` instance.

    If you have a ``datetime.datetime`` instance, it's recommended to convert
    it to a ``datetime.date`` first. If you don't, :class:`DateField` will
    localize the ``datetime.datetime`` to the :ref:`default timezone
    <default-current-time-zone>` and converted it to a ``datetime.date``
    instance, removing its time component. This is true for both storage and
    comparison.
```
<details><summary>Image</summary>

![image](https://github.com/django/django/assets/42296566/3ab3363e-b505-45ed-9e6f-7eb1244f20ff)
</details>

Reply:
No worries! I used what you have here (and changed "converted" to "convert").

Reply:
Thank you ⭐